metricbeat/module/prometheus/collector/collector_test.go

```diff
@@ -5,71 +5,138 @@ package collector
 import (
 	"testing"
 
-	"github.com/stretchr/testify/assert"
-
 	"github.com/elastic/beats/libbeat/common"
+	"github.com/golang/protobuf/proto"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/stretchr/testify/assert"
 )
 
-func TestDecodeLine(t *testing.T) {
+func TestGetPromEventsFromMetricFamily(t *testing.T) {
+	labels := common.MapStr{
+		"handler": "query",
+	}
 	tests := []struct {
-		Line  string
-		Event PromEvent
+		Family *dto.MetricFamily
+		Event  PromEvent
 	}{
 		{
-			Line: `http_request_duration_microseconds{handler="query",quantile="0.99"} 17`,
+			Family: &dto.MetricFamily{
+				Name: proto.String("http_request_duration_microseconds"),
+				Help: proto.String("foo"),
+				Type: dto.MetricType_COUNTER.Enum(),
+				Metric: []*dto.Metric{
+					{
+						Label: []*dto.LabelPair{
+							{
+								Name:  proto.String("handler"),
+								Value: proto.String("query"),
+							},
+						},
+						Counter: &dto.Counter{
+							Value: proto.Float64(10),
+						},
+					},
+				},
+			},
 			Event: PromEvent{
-				key:       "http_request_duration_microseconds",
-				value:     int64(17),
-				labelHash: `handler="query",quantile="0.99"`,
-				labels: common.MapStr{
-					"handler":  "query",
-					"quantile": 0.99,
+				key: "http_request_duration_microseconds",
```

Comment:
In a future step we could also extract the "units" part like `microseconds` as I assume this is also part of the convention.